### PR TITLE
don't run in timeouts on nightly that has no account_menu anymore

### PR DESCRIPTION
**Pull request by evgeni**
this does explicitly not assert the username, as for some reason
webdriver says .text is empty and I didn't dig deeper

at least we know we're logged in as some user

### CODE REVIEW

File: tests/test_pages.py [hunk 1]
```diff
@@ -50,10 +51,15 @@ def test_menu_item(selenium, user, url):
...
 
-    account_menu = WebDriverWait(selenium, 10).until(
-        EC.presence_of_element_located((By.ID, 'account_menu'))
-    )
-    assert account_menu.text == user.name, 'Logged in user shows the correct name'
+    try:
+        account_menu = WebDriverWait(selenium, 10).until(
+            EC.presence_of_element_located((By.ID, 'account_menu'))
+        )
+        assert account_menu.text == user.name, 'Logged in user shows the correct name'
```
Comment (ekohl) [inline tests/test_pages.py:58]: Shouldn't this assert be outside of the `try/except`?

**Comment by evgeni** [comment]
FWIW, the foreman commit that removed that is https://github.com/theforeman/foreman/commit/299c2976888bce605d29bbecbc5e4cab4cfb9fc5

File: tests/test_pages.py [hunk 2]
```diff
@@ -50,10 +51,15 @@ def test_menu_item(selenium, user, url):
     password_field.send_keys(user.password)
     password_field.submit()
 
-    account_menu = WebDriverWait(selenium, 10).until(
-        EC.presence_of_element_located((By.ID, 'account_menu'))
-    )
-    assert account_menu.text == user.name, 'Logged in user shows the correct name'
+    try:
+        account_menu = WebDriverWait(selenium, 10).until(
+            EC.presence_of_element_located((By.ID, 'account_menu'))
```
Comment (ekohl) [inline tests/test_pages.py:56]: Should we be looking at the future and first try the new user nav and then fall back to Foreman 2.4 and older?

File: tests/test_pages.py (hunk 1 above)
Comment (evgeni) [reply tests/test_pages.py:58]: It should, but as I've written in the commit message, `account.text` ends up empty, for some reason I didn't bother to debug.

File: tests/test_pages.py (hunk 2 above)
Comment (evgeni) [reply tests/test_pages.py:56]: That's a good idea.

File: tests/test_pages.py [hunk 3]
```diff
@@ -50,10 +51,15 @@ def test_menu_item(selenium, user, url):
     password_field.send_keys(user.password)
     password_field.submit()
 
-    account_menu = WebDriverWait(selenium, 10).until(
-        EC.presence_of_element_located((By.ID, 'account_menu'))
-    )
-    assert account_menu.text == user.name, 'Logged in user shows the correct name'
+    try:
```
Comment (ekohl) [inline tests/test_pages.py:54]:
```suggestion
    # Foreman 2.5 changed the navigation
    try:
```

File: tests/test_pages.py (hunk 1 above)
Comment (evgeni) [reply tests/test_pages.py]: documented that as an explicit fixme

File: tests/test_pages.py [hunk 4]
```diff
@@ -50,10 +51,18 @@ def test_menu_item(selenium, user, url):
...
+        account_menu = WebDriverWait(selenium, 10).until(
+            EC.presence_of_element_located((By.CLASS_NAME, 'user-nav-item'))
+        )
+    except TimeoutException:
+        account_menu = WebDriverWait(selenium, 10).until(
+            EC.presence_of_element_located((By.ID, 'account_menu'))
+        )
+        # FIXME: this assert should be *outside* the try/except, so it asserts on both
+        # kinds of user menu, but for some reason the new one (user-nav-item) ends up
+        # having account_menu.text an empty string, and thus the assert fails.
```
Comment (ekohl) [inline tests/test_pages.py:64]: I suspect that within the menu there is another item that has the actual text.

**Comment by evgeni** [comment]
On nightly, the structure is:

```html
<div class="pf-c-page__header-tools-item header-tool-item-hidden-lg user-nav-item">
   <div class="pf-c-dropdown pf-m-align-right" data-ouia-component-type="PF4/Dropdown" data-ouia-safe="true" data-ouia-component-id="OUIA-Generated-Dropdown-1">
      <button data-ouia-component-type="PF4/DropdownToggle" data-ouia-safe="true" data-ouia-component-id="OUIA-Generated-DropdownToggle-1" id="pf-dropdown-toggle-id-6" class="pf-c-dropdown__toggle pf-m-plain" type="button" aria-expanded="false" aria-haspopup="true">
         <span class="pf-c-dropdown__toggle-text">
            <svg style="vertical-align: -0.125em;" fill="currentColor" height="1em" width="1em" viewBox="0 0 512 512" aria-hidden="true" role="img" class="user-icon">
               <path d="M256 288c79.5 0 144-64.5 144-144S335.5 0 256 0 112 64.5 112 144s64.5 144 144 144zm128 32h-55.1c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16H128C57.3 320 0 377.3 0 448v16c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48v-16c0-70.7-57.3-128-128-128z"></path>
            </svg>
            Admin User
         </span>
         <span class="pf-c-dropdown__toggle-icon">
            <svg style="vertical-align: -0.125em;" fill="currentColor" height="1em" width="1em" viewBox="0 0 320 512" aria-hidden="true" role="img">
               <path d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"></path>
            </svg>
         </span>
      </button>
   </div>
</div>
```

While before that, it was:

```html
<a id="account_menu" role="button" class="nav-item-iconic dropdown-toggle" aria-haspopup="true" aria-expanded="false" href="#"><span aria-hidden="true" class="fa fa-user avatar small"></span>Evgeni Golov <span class="caret"></span></a>
```

And yeah, I also tried selecting on `pf-c-dropdown__toggle-text`, which didn't have anything useful either :/

**Comment by evgeni** [comment]
and @tbrisker found a nice way to solve this, so now with 100% assert

**Review by ekohl — APPROVED** [review]

**Pull request merged**